### PR TITLE
[CDAP-8021] Preferences for application and programs (Confirmation & error message)

### PR DESCRIPTION
**Pull request by tonybach**
JIRA: https://issues.cask.co/browse/CDAP-8021?focusedCommentId=27080&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-27080

- Added confirmation message when user saves their preferences on the application and program levels.
- Repositioned error message.

### CODE REVIEW

**Comment by tonybach** [comment]
Merging the other preferences PR caused a bunch of conflicts, so I'm resolving those right now.

UPDATE: Conflicts resolved, but accidentally squashed the previous commits.

**Comment by tonybach** [comment]
Working on improvements:

- [ ] There should be no side scrolling in the modal
- [ ]  Add visual treatment when the field is in focus
- [ ]  [Save & Close] should not be disabled when there is a blank row. Button is disabled only when only one text input (either key or value) is filled and the other is not.
- [ ]  if Namespace is too long for the size of the modal, name is truncated, ellipsis are added, user can view name by hovering.
- [ ] Preference title: please add colon (:) before App name
- [ ] Set lengths of App name by the width of the modal. Truncate and add ellipsis if needed.
- [ ] Add confirmation message after user "resets" successfully. (display for 3 seconds before fading)
- [ ] Add rows count
- [ ] [Button] is displayed in grey when disabled

File: cdap-ui/app/cdap/components/EntityCard/EntityCardHeader/index.js [hunk 1]
```diff
@@ -44,16 +44,38 @@ export default class EntityCardHeader extends Component {
...
-      >
-        <h4>
-          <span className={classNames('entity-icon', this.props.entity.icon)}></span>
-          <span className="entity-type">
-            {T.translate(`commons.entity.${this.getType()}.singular`)}
-          </span>
-        </h4>
+      <div className="card-header-wrapper">
+        {
+          this.props.preferencesSaved ?
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/EntityCard/EntityCardHeader/index.js]: We use EntityCard for all the entities in CDAP. A prop named `preferencesSaved` might not be the right name for it. May be `successMessage`?

File: cdap-ui/app/cdap/components/EntityCard/FastActions/index.js [hunk 2]
```diff
@@ -49,6 +49,11 @@ export default class FastActions extends Component {
   onSuccess(action) {
     if (action === 'startStop') { return; }
 
+    if (action === 'setPreferences') {
+      this.props.onPreferencesSaved();
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/EntityCard/FastActions/index.js]: As mentioned above this should have been `this.props.onSuccess()`. Preference as it is a FastAction. We should not have anything outside the scope of `FastAction` here as it is going to be the base for all FastActions.

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**